Mathlib/Algebra/Module/Zlattice.lean

```diff
@@ -409,23 +426,31 @@ theorem Zlattice.FG : AddSubgroup.FG L := by
     exact fg_span (LinearIndependent.setFinite h_lind)
 
 theorem Zlattice.module_finite : Module.Finite ℤ L :=
-  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K hs))
+  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K L))
+
+instance instModuleFinite_Zlattice {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+    [FiniteDimensional ℝ E] [ProperSpace E] (L : AddSubgroup E) [DiscreteTopology L]
+    [IsZlattice ℝ L] : Module.Finite ℤ L := Zlattice.module_finite ℝ L
 
 theorem Zlattice.module_free : Module.Free ℤ L := by
```

Comment:
Here again, this is probably true without the lattice instance, being discrete should be enough.

Reply:
Yes, that's a very good point. The class `IsZlattice` is for full lattices but these results are true for lattices in general. I'll fix that. Thanks!

Reply:
Ok. I pushed a new version without the `IsZlattice` hypothesis.